src/usermod.c

```diff
@@ -443,11 +441,7 @@ static char *new_pw_passwd (char *pw_pass)
 		              "updating password", user_newname, user_newid, 0);
 #endif
 		SYSLOG ((LOG_INFO, "unlock user '%s' password", user_newname));
-		s = pw_pass;
-		while ('\0' != *s) {
-			*s = *(s + 1);
-			s++;
-		}
+		memmove(pw_pass, pw_pass + 1, strlen(pw_pass) + 1);
```

Comment:
hm, pw_pass+1 is not right for the length, is it?  You're starting at pw_pass + 1, so you'll be accessing one byte past the end of the \0 ending pw_pass.

Reply:
You're right, due to omission of temporary variable I was counting the wrong length, thanks for noticing.